core/rfb.js

```diff
@@ -1060,15 +1062,27 @@ export default class RFB extends EventTargetMixin {
         let pos = clientToElement(ev.clientX, ev.clientY,
                                   this._canvas);
 
+        /* Map mouse back and forward mouse buttons (3 and 4) to 7 and 8
+         *
+         * NOTE: This only works on chromium-based browsers. There is
+         *       no support for firefox/safari.
```

Comment:
Strictly speaking, Safari has buggy support rather than no support. Right?

Reply:
Correct.

I just realized that safari send the incorrect `ev.button` value, but the button mask `ev.buttons` seems to work correctly. With back pressed, `ev.buttons` is 8, and with forward pressed, it is 16.

We could look at those instead of keeping mouse button state ourselves?